src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -150,9 +150,14 @@ public StatementRestrictions(StatementType type,
          *     allow two IN for the same entity but that doesn't seem very useful)
          *   - The value_alias cannot be restricted in any way (we don't support wide rows with indexed value
          *     in CQL so far)
+         *   - CONTAINS and CONTAINS_KEY cannot be used with UPDATE or DELETE
          */
         for (Relation relation : whereClause.relations)
         {
+            if ((relation.isContains() || relation.isContainsKey()) && (type.isUpdate() || type.isDelete()))
+            {
+                throw invalidRequest("Cannot use %s with %s", type, relation.operator());
```

Comment:
Placing this new validation on first place produces changes in the error message that is thrown in some queries that were also correctly rejected before, as it can see in the modified tests. For example:
```
CREATE TABLE t (pk int, ck int, v int, PRIMARY KEY (pk, ck));
UPDATE t SET value = ? WHERE pk CONTAINS ? AND ck = ?
```
Will throw the new message instead of the previous `Cannot use CONTAINS on non-collection column pk`.

Both options are correct, but I think that throwing the new query-level generic error message is preferable to throwing the column-level error messages that were previously thrown. Thus, I think that placing this check on the first place is the best approach, although it changes behaviour a bit. @blerer wdyt?

Reply:
I agree. This message makes more sense.